Mathlib/RingTheory/KrullDimension.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fangming Li, Jujian Zhang
+-/
+import Mathlib.AlgebraicGeometry.PrimeSpectrum.Basic
```

Comment:
This should import `Mathlib.RingTheory.PrimeSpectrum` and results that rely on Zariski topology should be moved to `AlgebraicGeometry`.

Reply:
But in the current version of Mathlib of this pull request, there is no `Mathlib.RingTheory.PrimeSpectrum`. Do I need to run `git merge origin/master`?

Reply:
It's good practice to merge master if it has been a week or more. Eg, GitHub can't tell if there is a linter failure in your PR because of new changes to master.

Reply:
But after I ran `git merge origin/master`, there is still no `Mathlib.RingTheory.PrimeSpectrum`. How can I solve the problem?

Reply:
After I changed the import into `Mathlib.RingTheory.PrimeSpectrum`, the core definition `ringKrullDim` stopped compiling because the instance `Preorder (PrimeSpectrum R)` could not be synthesized. And all the contents of the file rely on the definition `ringKrullDim`. I'm not sure how to resolve the problem now?

Reply:
The current preorder on Spec R is given by the specialization order and depends on the topology. If it is easy to do you can redefine it using the induced order from ideals and move it to `RingTheory`. If not this could be done in a separate PR. But please still split the lemmas really depending on the topology into a new PR.

Reply:
:/ Let's move this file to `AlgebraicGeometry.KrullDimension` and follow up with another PR moving back into `RingTheory` with the standard preorder and splitting off `Field` results.

Reply:
So for this PR, shall I create a file `AlgebraicGeometry/KrullDimension.lean`, or shall I create a folder `AlgebraicGeometry/KrullDimension` and then a file `AlgebraicGeometry/KrullDimension/Basic.lean`?

Reply:
Just a single file is ok for now since it will be touched again in another PR. We should get the correct preorder instance before splitting

Reply:
Or we can wait for #15108.

Reply:
> Or we can wait for #15108.

Yeah I feel that this is a good idea. Also I am not sure whether redefining `ringKrullDim` is a good idea because @jjaassoonn and I have already formalized a bunch of results (e.g. the Krull dimensions of PIDs, Artinian rings, etc.) based on the definition
```
noncomputable def ringKrullDim (R : Type*) [CommRing R] : WithBot (WithTop ℕ) :=
  krullDim (PrimeSpectrum R)
```
and I'm not sure if our proofs will remain as concise as before after redefining `ringKrullDim`.

Reply:
I don't think `ringKrullDim` needs to change. We just need the preorder instance in scope for it to work. I didn't realize it was already the usual partial order (ideal containment) and can just be moved easily.

Reply:
> Or we can wait for #15108.

This will be on master shortly

Reply:
I have merged with master and changed the import into `Mathlib.RingTheory.PrimeSpectrum`, but `Preorder (PrimeSpectrum R)` still cannot be synthesized?

Reply:
I just merged master for you. Please try again.

Reply:
(and yes next time please run `git fetch` before merging and merge with `origin master` and not the `master` branch on your local machine. Or `git pull origin master` works too I think.)

Reply:
The theorem “ringKrullDim_le_of_surjective” stopped compiling after I changed the import because the proof of the theorem uses “PrimeSpectrum.comap”. But I don’t think it is reasonable to move “ringKrullDim_le_of_surjective” to “AlgebraicGeometry”, because the theorem is one of the most basic theorems about the Krull dimension of a ring. Is there any good way to solve this problem?

Reply:
The easy way out is to replace `comap` with its definition and the lemma with `Ideal.comap_injective_of_surjective` in the proof. The long term solution is to unbundle `comap` from continuity and move it to `RingTheory`

Reply:
There should be a `PrimeSpectrum.comapFun`.

Reply:
#15114 breaks out the function. It was done a bit hastily so please look it over.

Reply:
@mattrobball Should #15114 be a dependency of this PR?

Reply:
Assuming it looks fine yes. Otherwise, I would like to find a reasonable way to keep the line between commutative algebra and algebraic geometry at importing the Zariski topology.